quantum/pointing_device/pointing_device_modes.h

```diff
@@ -0,0 +1,164 @@
+void pointing_mode_key_momentary(uint8_t mode_id, bool pressed); // momentary change of pointing mode while button is held
+void pointing_mode_key_toggle(uint8_t mode_id, bool pressed);    // toggle pointing mode on/off on keypress
+
+/* ----------For custom pointing modes---------------------------------------------------------------------------- */
+void              set_pointing_mode(pointing_mode_t pointing_mode);                                          // overwrite local pointing_mode status
+void              pointing_mode_update(void);                                                                // update direction & divisor from current mode id, x, y
+mouse_xy_report_t apply_divisor_xy(int16_t value);                                                           // divide value by current divisor and limit value to x/y range
+int8_t            apply_divisor_hv(int16_t value);                                                           // divide value by current divisor and limit value to h/v range
+int16_t           multiply_divisor_xy(mouse_xy_report_t value);                                              // multiply mouse x/y value by current divisor
+int16_t           multiply_divisor_hv(int8_t value);                                                         // multiply mouse h/v value by current divisor
+void              pointing_tap_codes(uint16_t kc_left, uint16_t kc_down, uint16_t kc_up, uint16_t kc_right); // turn pointing_mode x/y values into keycode taps
+
+/* ----------For multiple pointing devices------------------------------------------------------------------------ */
+#if defined(SPLIT_POINTING_ENABLE) && defined(POINTING_DEVICE_COMBINED)
+bool is_pointing_mode_on_left(void);   // check if left is current active pointing mode side
+void pointing_mode_switch_hands(void); // change witch hand is active on pointing mode
+#endif
+
+/* ----------Callbacks for modifying and adding pointing modes---------------------------------------------------- */
+bool process_pointing_mode_kb(pointing_mode_t pointing_mode, report_mouse_t* mouse_report);   // keyboard level
+bool process_pointing_mode_user(pointing_mode_t pointing_mode, report_mouse_t* mouse_report); // user/keymap level
+
+/* ----------Callbacks for adding/modifying pointing device mode divisors----------------------------------------- */
+uint8_t get_pointing_mode_divisor_kb(uint8_t mode_id, uint8_t direction);   // adding divisors at keyboard level
+uint8_t get_pointing_mode_divisor_user(uint8_t mode_id, uint8_t direction); // adding divisors at user/keymap level
+bool    pointing_mode_divisor_postprocess_kb(uint8_t* divisor);             // modifying divisors at keyboard level
+bool    pointing_mode_divisor_postprocess_user(uint8_t* divisor);           // modifying divisors at user/keymap level
+void    pointing_mode_divisor_override(uint8_t divisor);                    // Override current divisor until next update
```

Comment:
The intent was to allow for modification of divisors after they were updated (allowing for for modifiers such as the precision mode that would act as a multiplier for divisors when active, or for dynamic defaults etc.).

The divisor override was to allow for changing divisors between updates and still have post processes applied.

This originally was used to allow for differential horizontal and vertical drag scroll divisors but I found from testing that this did not play nicely with elements that treated horizontal and vertical scroll the same so I removed it.

Admittedly this is pretty niche and not used by default so it could be cut.

Reply:
My intention was it'd allow you to easily enable a pointing mode on both sides simultaneously. Sure there are other ways to achieve this already. 

Anyway I missed this would need updating to allow for that.
```c
    if (is_pointing_mode_on_left()) {
        left_report = pointing_device_modes_task(left_report);
    } else {
        right_report = pointing_device_modes_task(right_report);
    }
```